### PR TITLE
Smoothing maps and record

### PR DESCRIPTION
**Pull request by Jeremmmyyyyy**
This PR will : 
- Fix navigation bugs when going from home to maps when clicking on a path. The camera will be centered on the selected path, display it as if it was clicked and show the pop up.
- Allow to click on an itinerary in the search bar so that it navigates directly to it's map overview.
- Navigating between home and Maps now resets the stored states and is not crashing anymore.
- Now clicking on the map in the map overview unselect the path if there was one selected and displayed in bold.
- Add spot was changed in order to match the figma design.
- Tests are fixed.
- Now clicking on the pop up of a path opens the path overview with all the spots in full screen width.
- Clicking on one of these pins opens the pin description and shows the picture.

<image src = https://github.com/EPFL-SwEnt-2024-LaStartUp/TripTracker/assets/79469048/02ed1fbf-7684-4099-a8bf-1ca1e6763282 width = 300> <image src = https://github.com/EPFL-SwEnt-2024-LaStartUp/TripTracker/assets/79469048/12eb7fcd-7802-44c6-b476-cde45bff0d99 width = 300>

### CODE REVIEW

File: app/src/main/java/com/example/triptracker/MainActivity.kt
Comment (theoschiff) [inline app/src/main/java/com/example/triptracker/MainActivity.kt:1]: very clever fix to have a default id for when you do not click on an itinerary

File: app/src/main/java/com/example/triptracker/view/Navigation.kt [hunk 1]
```diff
@@ -54,6 +55,7 @@ class Navigation(val navController: NavHostController) {
   fun navigateTo(destination: TopLevelDestination) {
     navController.navigate(destination.route) {
       currentDestination = destination
+      navController.currentBackStackEntry?.arguments?.putString("id", "")
```
Comment (theoschiff) [inline app/src/main/java/com/example/triptracker/view/Navigation.kt:58]: great fix for the back and forth problem. Could you add a quick comment to explain why you made this choice?

File: app/src/main/java/com/example/triptracker/view/home/HomeScreen.kt [hunk 2]
```diff
@@ -144,12 +144,7 @@ fun HomeScreen(navigation: Navigation, homeViewModel: HomeViewModel = viewModel(
                     DisplayItinerary(
                         itinerary = itinerary,
                         navigation = navigation,
-                        onClick = {
-                          navigation.navigateTo(navigation.getTopLevelDestinations()[1])
-                          navigation.navController.navigate(
-                              Route.MAPS +
-                                  "/${itinerary.location.latitude}/${itinerary.location.longitude}")
-                        })
+                        onClick = { navigation.navigateTo(Route.MAPS, itinerary.id) })
```
Comment (theoschiff) [inline app/src/main/java/com/example/triptracker/view/home/HomeScreen.kt:147]: much nicer!! good job

File: app/src/main/java/com/example/triptracker/view/home/HomeScreen.kt [hunk 3]
```diff
@@ -275,7 +270,10 @@ fun SearchBarImplementation(
         items(listToShow) { itinerary ->
           ItineraryItem(
               itinerary = itinerary,
-              onItineraryClick = { /* TODO: Implement navigation to itinerary details */})
+              onItineraryClick = {
+                navigation.navigateTo(navigation.getTopLevelDestinations()[1])
+                navigation.navController.navigate(Route.MAPS + "/${itinerary.id}")
```
Comment (theoschiff) [inline app/src/main/java/com/example/triptracker/view/home/HomeScreen.kt:275]: This seems odd. Could you please recheck this, it should be the same as above.

File: app/src/main/java/com/example/triptracker/view/map/AddSpot.kt
Comment (theoschiff) [inline app/src/main/java/com/example/triptracker/view/map/AddSpot.kt:1]: Changes in UI. Generic colors. LGTM

File: app/src/main/java/com/example/triptracker/view/map/MapPopups.kt
Comment (theoschiff) [inline app/src/main/java/com/example/triptracker/view/map/MapPopups.kt:1]: Great feature to now be able to click on the PathItem.

File: app/src/main/java/com/example/triptracker/view/map/RecordScreen.kt [hunk 4]
```diff
@@ -276,45 +277,43 @@ fun Map(
     if (!viewModel.isInDescription()) {
       StartWindow(viewModel = viewModel, context)
 
-      if (!viewModel.addSpotClicked.value) {
```
Comment (theoschiff) [inline app/src/main/java/com/example/triptracker/view/map/RecordScreen.kt:279]: why did you remove this check? not saying that it's wrong but I don't understand why it's not present anymore

File: app/src/main/java/com/example/triptracker/view/map/RecordScreen.kt [hunk 5]
```diff
@@ -424,23 +425,23 @@ fun Map(
...
-                                  color = Color.White,
+                                  color = md_theme_light_onPrimary,
                                   fontSize = 16.sp,
                                   fontFamily = Montserrat,
                                   fontWeight = FontWeight.Normal,
                                   lineHeight = 10.sp),
                           colors =
                               OutlinedTextFieldDefaults.colors(
-                                  unfocusedTextColor = Color.White,
+                                  unfocusedTextColor = md_theme_light_onPrimary,
```
Comment (theoschiff) [inline app/src/main/java/com/example/triptracker/view/map/RecordScreen.kt:435]: thanks for changing to generic colours!

File: app/src/main/java/com/example/triptracker/view/map/MapOverview.kt
Comment (theoschiff) [inline app/src/main/java/com/example/triptracker/view/map/MapOverview.kt:1]: Great fixes here. LGTM

File: app/src/main/java/com/example/triptracker/viewmodel/MapViewModel.kt [hunk 6]
```diff
@@ -27,12 +29,17 @@ import kotlinx.coroutines.launch
...
     val geocoder: NominatimApi = NominatimApi(),
-    private val pathList: MutableLiveData<ItineraryList> = MutableLiveData<ItineraryList>(),
+    val pathList: MutableLiveData<ItineraryList> = MutableLiveData<ItineraryList>(),
     private val repository: ItineraryRepository = ItineraryRepository(),
     val filteredPathList: MutableLiveData<Map<Itinerary, List<LatLng>>> =
         MutableLiveData<Map<Itinerary, List<LatLng>>>()
 ) : ViewModel() {
 
+  val DUMMY_SELECTED_POLYLINE =
+      SelectedPolyline(
```
Comment (theoschiff) [inline app/src/main/java/com/example/triptracker/viewmodel/MapViewModel.kt:39]: Could you please add a comment here on why this dummy_selected_polyline is necessary?
Thanks

**Review by theoschiff — APPROVED** [review]
Really neat PR. Excellent fixes and new features. 
I approved it as there are only minor fixes but please check my comments before merging

File: app/src/main/java/com/example/triptracker/view/map/RecordScreen.kt (hunk 4 above)
Comment (Jeremmmyyyyy) [reply app/src/main/java/com/example/triptracker/view/map/RecordScreen.kt:279]: because it was precised on the figma that we can have the center location button

File: app/src/main/java/com/example/triptracker/viewmodel/MapViewModel.kt (hunk 6 above)
Comment (Jeremmmyyyyy) [reply app/src/main/java/com/example/triptracker/viewmodel/MapViewModel.kt:39]: This is to avoid nullability issues ! I added a comment thanks !
